daemon/container_operations_unix.go

```diff
@@ -320,6 +323,9 @@ func (daemon *Daemon) setupConfigDir(c *container.Container) (setupErr error) {
 		if err := os.Chown(fPath, rootIDs.UID+uid, rootIDs.GID+gid); err != nil {
 			return errors.Wrap(err, "error setting ownership for config")
 		}
+		if err := os.Chmod(fPath, configRef.File.Mode); err != nil {
```

Comment:
There's a lot of duplication between secrets and configs here; looks like that's being addressed in https://github.com/moby/moby/pull/33702